Bug fixes & refactor

jkilzi opened the pull request:
## Bug Fixes:
- Sorting by `last_reported` doesn't work
- Sorting is performed only on the client-side
- Search results are not displayed when using the filters from a page other than the first one
- Clusters search don't work properly (https://issues.redhat.com/browse/RHIN-1715) 

### Before 
https://github.com/user-attachments/assets/993d63a0-aa16-4c3b-b331-b2caf182eb83

## Changelog:
- Bumps Backstage to 1.29.2
- Removes @backstage-plugins/resource-optimizations-react, the useful bits of this package are now part of the front-end plugin.
- Enables the plugin to be started in dev-mode with minimal dependencies using the `yarn start:dev` script. The `dev` script is now called `start`.
- Adds generated api-reports
- Introduces types for the request parameters: `GetRecommendationListRequest` and `GetRecommendationByIdRequest`
- Restrcutures the code in the FE plugin based on the heirarchical structure of the components, emphazising public, and private modules.
- Introduces the new index page, including the bug-fixes detailed above
- Patches the openapi.yaml to generate enums for the `order_how` and `order_by` query parameters
- Renames the `OptimizationsClient` to `OptimizationsApiClientProxy` to make it's purpose more explicit.

### After
https://github.com/user-attachments/assets/7458574a-825d-4e7a-b4ed-497f569b7792

## To-dos
- Deprecate `ExampleComponent.tsx` and co.

jkilzi committed:
WIP @ resource-optimization/fixes

Signed-off-by: Jonathan Kilzi <jkilzi@redhat.com>

jkilzi committed:
updates api-reports

Signed-off-by: Jonathan Kilzi <jkilzi@redhat.com>

jkilzi committed:
fixes the presentation in the last_reported column

Signed-off-by: Jonathan Kilzi <jkilzi@redhat.com>

jkilzi committed:
after yarn dedupe

Signed-off-by: Jonathan Kilzi <jkilzi@redhat.com>

jkilzi committed:
fixes code style issues

Signed-off-by: Jonathan Kilzi <jkilzi@redhat.com>

jkilzi committed:
dedupe packages

Signed-off-by: Jonathan Kilzi <jkilzi@redhat.com>

jkilzi committed:
fix code-style

Signed-off-by: Jonathan Kilzi <jkilzi@redhat.com>

jkilzi committed:
Renames serializer.mjs to yaml.mjs

Signed-off-by: Jonathan Kilzi <jkilzi@redhat.com>

PreetiW commented:
@jkilzi the after video, I can see the dates are not in the `days ago` format, is it fixed in the code? (FYI, I haven't run the code locally)

jkilzi commented:
> @jkilzi the after video, I can see the dates are not in the `days ago` format, is it fixed in the code? (FYI, I haven't run the code locally)

Yes! I addressed this issue later, here is the commit: bdf478ee99cfcffe2a01ad2987672725f4dc32e8

https://github.com/user-attachments/assets/809f0105-d2ef-431d-9c09-430709441870

jkilzi committed:
Bumps Backstage to 1.29.2

See: https://backstage.io/docs/releases/v1.28.0

Signed-off-by: Jonathan Kilzi <jkilzi@redhat.com>

jkilzi committed:
Fixes the "publish check" CI step failure

Signed-off-by: Jonathan Kilzi <jkilzi@redhat.com>

PreetiW commented on workspaces/resource-optimization/plugins/resource-optimization-common/src/generated/apis/DefaultApi.client.ts:
```diff
@@ -112,8 +112,14 @@ export class DefaultApiClient {
+          | 'workload'
+          | 'container'
+          | 'last_reported';
+        orderHow?: 'asc' | 'desc';
```
There is issue because of this line https://github.com/jkilzi/backstage-community-plugins/blob/18fdf73d13948543493e7ad4d160cc37b28c7215/workspaces/resource-optimization/plugins/resource-optimization-common/src/generated/apis/DefaultApi.client.ts#L74 
it is expecting `recommendationId` and we are passing same from frontend but it is converting it to `recommendation_id`. Please refer the screenshot for reference 
![image](https://github.com/user-attachments/assets/d04f4479-5945-4018-9aac-dffa56509aea)

we had faced the similar issue before fixed it but I am not sure if something got changed in recent commits. @jkilzi can you please check and fix this? Thanks!

jkilzi replied:
OK, I see what's the issue now.

Again, we need to allow the function that handles the case conversion to make an axception for the `recommendationId` path parameter so it doesn't convert it to `recommendation_id`.

y0n1 replied:
Addressed in https://github.com/jkilzi/backstage-community-plugins/commit/0ac0c4e0b37b27394a1330cba949d1321922b3e2

https://github.com/user-attachments/assets/caf52d1e-17cc-45d5-bad0-1735bbe612e1

PreetiW reviewed (changes requested):
minor comment for detail page, please check 🙌🏻 once that is fixed the PR is good to merge 🚀 
Thank you @jkilzi for the PR and fixes 🙌🏻

jkilzi committed:
bug fixes

- Solves an issue related to the recommendation plotData object where the object keys were transformed incorrectly
- Bumps @y0n1/json to 2.0.0
- Adds msw for writing tests for the client proxy module

Signed-off-by: Jonathan Kilzi <jkilzi@redhat.com>

PreetiW reviewed (approved):
Thank you for fixing the issue with response, everything is working fine now. LGTM 🚀
